[Bug]: aws_elasticache_replication_group panics with runtime error: index out of range [0] with length 0

### Terraform Core Version

v1.8.4

### AWS Provider Version

v5.49.0

### Affected Resource(s)

```
│   with module.gitlab_ref_arch_aws.aws_elasticache_replication_group.gitlab_redis[0],
│   on ../../../modules/gitlab_ref_arch_aws/elasticache.tf line 55, in resource "aws_elasticache_replication_group" "gitlab_redis":
│   55: resource "aws_elasticache_replication_group" "gitlab_redis" {
``` 

### Expected Behavior

no crash

### Actual Behavior

it crash

### Relevant Error/Panic Output Snippet

```shell
Stack trace from the terraform-provider-aws_v5.49.0_x5 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 380 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/elasticache.resourceReplicationGroupRead({0x1ca85a28, 0xc003504d80}, 0xc002e3b300, {0x1c829140?, 0xc002431d40?})
	github.com/hashicorp/terraform-provider-aws/internal/service/elasticache/replication_group.go:653 +0x1e25
github.com/hashicorp/terraform-provider-aws/internal/provider.New.(*wrappedResource).Read.interceptedHandler[...].func9(0xc002e3b300?, {0x1c829140?, 0xc002431d40})
	github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:113 +0x283
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x1ca85a28?, {0x1ca85a28?, 0xc0033dced0?}, 0xd?, {0x1c829140?, 0xc002431d40?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/resource.go:790 +0x7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc001641c00, {0x1ca85a28, 0xc0033dced0}, 0xc001234c30, {0x1c829140, 0xc002431d40})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/resource.go:1089 +0x529
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc003ff87f8, {0x1ca85a28?, 0xc0033dce10?}, 0xc002fbdbc0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/grpc_provider.go:667 +0x4aa
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ReadResource(0xc0016871f0, {0x1ca85a28?, 0xc0033dcb40?}, 0xc002fbdbc0)
	github.com/hashicorp/terraform-plugin-mux@v0.15.0/tf5muxserver/mux_server_ReadResource.go:35 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc0006d1b80, {0x1ca85a28?, 0xc0033dc3c0?}, 0xc002ef47e0)
	github.com/hashicorp/terraform-plugin-go@v0.22.2/tfprotov5/tf5server/server.go:775 +0x4c3
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x1c5e8f00, 0xc0006d1b80}, {0x1ca85a28, 0xc0033dc3c0}, 0xc002e3b080, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.22.2/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:482 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000ebe200, {0x1ca85a28, 0xc0033dc330}, {0x1cad0240, 0xc000548000}, 0xc001f158c0, 0xc0024d31d0, 0x256f9708, 0x0)
	google.golang.org/grpc@v1.63.2/server.go:1369 +0xdf8
google.golang.org/grpc.(*Server).handleStream(0xc000ebe200, {0x1cad0240, 0xc000548000}, 0xc001f158c0)
	google.golang.org/grpc@v1.63.2/server.go:1780 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 55
	google.golang.org/grpc@v1.63.2/server.go:1030 +0x125

Error: The terraform-provider-aws_v5.49.0_x5 plugin crashed!
```
```


### Terraform Configuration Files

configuration from https://gitlab.com/gitlab-org/gitlab-environment-toolkit

### Steps to Reproduce

terraform apply was killed while creating the elasticache, subsequent terraform apply then crash

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

if I try do delete the cluster in aws UI, I have the following error 

> All resources in replication group must be in a valid state for deletion in order to delete the replication group

### References

_No response_

### Would you like to implement a fix?

None

--- a/.changelog/38797.txt
+++ b/.changelog/38797.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_elasticache_replication_group: Fix crash when setting `replicas_per_node_group` if node groups are empty
+```

--- a/internal/service/elasticache/replication_group.go
+++ b/internal/service/elasticache/replication_group.go
@@ -682,7 +682,9 @@ func resourceReplicationGroupRead(ctx context.Context, d *schema.ResourceData, m
 	}
 
 	d.Set("num_node_groups", len(rgp.NodeGroups))
-	d.Set("replicas_per_node_group", len(rgp.NodeGroups[0].NodeGroupMembers)-1)
+	if len(rgp.NodeGroups) > 0 {
+		d.Set("replicas_per_node_group", len(rgp.NodeGroups[0].NodeGroupMembers)-1)
+	}
 
 	d.Set("cluster_enabled", rgp.ClusterEnabled)
 	d.Set("cluster_mode", rgp.ClusterMode)

